internal/provider/resource_aci_certificate_authority.go

```diff
@@ -0,0 +1,602 @@
+	CertChain     types.String `tfsdk:"cert_chain"`
+	Descr         types.String `tfsdk:"description"`
+	Name          types.String `tfsdk:"name"`
+	NameAlias     types.String `tfsdk:"name_alias"`
+	OwnerKey      types.String `tfsdk:"owner_key"`
+	OwnerTag      types.String `tfsdk:"owner_tag"`
+	TagAnnotation types.Set    `tfsdk:"annotations"`
+}
+
+// TagAnnotationPkiTPResourceModel describes the resource data model for the children without relation ships.
+type TagAnnotationPkiTPResourceModel struct {
+	Key   types.String `tfsdk:"key"`
+	Value types.String `tfsdk:"value"`
+}
+
+type PkiTPIdentifier struct {
+	Name types.String
+}
+
+func (r *PkiTPResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	tflog.Debug(ctx, "Start metadata of resource: aci_certificate_authority")
+	resp.TypeName = req.ProviderTypeName + "_certificate_authority"
+	tflog.Debug(ctx, "End metadata of resource: aci_certificate_authority")
+}
+
+func (r *PkiTPResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	tflog.Debug(ctx, "Start schema of resource: aci_certificate_authority")
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "The certificate_authority resource for the 'pkiTP' class",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "The distinguished name (DN) of the Certificate Authority object.",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"parent_dn": schema.StringAttribute{
+				Optional:            true,
+				MarkdownDescription: "The distinguished name (DN) of the parent object.",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"annotation": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					SetToStringNullWhenStateIsNullPlanIsUnknownDuringUpdate(),
+				},
+				Default:             stringdefault.StaticString(globalAnnotation),
+				MarkdownDescription: `The annotation of the Certificate Authority object.`,
+			},
+			"cert_chain": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					SetToStringNullWhenStateIsNullPlanIsUnknownDuringUpdate(),
+				},
+				MarkdownDescription: `The PEM-encoded chain of trust from the trustpoint to a trusted root authority.`,
+			},
+			"description": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					SetToStringNullWhenStateIsNullPlanIsUnknownDuringUpdate(),
```

Comment:
Do we test this functionality somewhere?

Reply:
Yes. The newer attributes work on older version of the APICs as well